ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/FourActivity.java

```diff
@@ -0,0 +1,31 @@
+package net.boondockradio.activitiesandintents;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+public class FourActivity extends AppCompatActivity {
+
+    private static final String KEY_MESSAGE = "message";
+
+    public static void startActivity(Context context) {
+        Intent intent = new Intent(context, FourActivity.class);
+        intent.putExtra(KEY_MESSAGE, "on moved 4");
+        context.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_four);
+
+        Intent intent = getIntent();
+
+        String message = intent.getStringExtra(KEY_MESSAGE);
+        TextView text = (TextView) findViewById(R.id.text);
+
+        text.setText(message);
```

Comment:
I did some research.
Its just a warning. 

`Android Studio is based on IntelliJ IDEA, and this is a feature of IntelliJ that gives you warnings at compile time when you are not checking if an object returned by a method is null before using it.`

We can be sure its not null because findViewById would of failed if it was.

4 things we can do to avoid the warning 
1. `@SuppressWarnings({"NullableProblems"})` before `onCreate`
2. `        assert text != null;
        text.setText(message);`
3. if statement
4. make text a member variable